sbol_utilities/component.py

```diff
@@ -519,3 +519,141 @@ def ed_protein(definition: str, **kwargs) -> sbol3.ExternallyDefined:
+    participations.append(modifier_participation)
+    
+    # part_in_backbone_seq = get reactant sequence
+    ds_reactant = Dseqrecord(part_in_backbone, linear=linear, circular=circular)
+
+    prefix, part, suffix = ds_reactant.cut(restriction_enzymes) #how do you extract the part if there are too many cuts?
+
+    part_sequence = part.seq
+    # product_component # How to define the product component? Do we want to use it afterwards?
+    part_comp, part_seq = component.dna_component_with_sequence(identity='part', sequence=str(part_sequence), **kwargs)
+    # product_subcomponent
+    part_subcomp = sbol3.SubComponent(part_comp)
+    
+    part_in_backbone_comp, part_in_backbone_seq = component.dna_component_with_sequence(identity='part_in_backbone', sequence=str(part_in_backbone))
+    part_in_backbone_subcomp = sbol3.SubComponent(part_in_backbone)
+    reactant_participation = sbol3.Participation(roles=[sbol3.SBO_REACTANT], participant=part_in_backbone_subcomp)
+    participations.append(reactant_participation)
+    product_participation = sbol3.Participation(roles=[sbol3.SBO_PRODUCT], participant=part_subcomp)
+    participations.append(product_participation)
+    
+    interaction = sbol3.Interaction(types=[sbol3.SBO_INHIBITION], participations=participations)
+                    
+    return tuple([part_comp,interaction])
+
+
+def ligation(reactants:List[sbol3.Component])-> Tuple[sbol3.Component, sbol3.Interaction]:
+    """Creates a Part Extract Component and a digestion interaction.
+
+    :param part_in_backbone: Part in backbone to be digested. 
+    :param restriction_enzymes: Restriction enzyme with correct DisplayID from Bio.Restriction as Externally Defined.
+    :param linear: Boolean to inform if the reactant is linear.
+    :param circular: Boolean to inform if the reactant is circular.
+    :param **kwargs: Keyword arguments of any other Component attribute.
+    :return: A tuple of Component and Interaction.
+    """
+    #search overhangs
+    #compare overhangs sequence
+    #2 matching overhangs creates a meets constrain
+    #create preceed constrain
+    #create composite part or part in backbone
+
+    return tuple([part_comp,interaction])
+
+    return 1
+
+class Assembly_plan_single_enzyme():
+    """Creates a Part Extract Component and a digestion interaction.
+
+    :param part_in_backbone: Part in backbone to be digested. 
+    :param restriction_enzymes: Restriction enzyme with correct DisplayID from Bio.Restriction as Externally Defined.
+    :param linear: Boolean to inform if the reactant is linear.
+    :param circular: Boolean to inform if the reactant is circular.
+    :param **kwargs: Keyword arguments of any other Component attribute.
+    :return: A tuple of Component and Interaction.
+    """
+
+    def __init__(self, parts_in_backbone: List(sbol3.Component), acceptor_backbone: sbol3.Component, restriction_enzyme: str):
```

Comment:
Is restriction enzyme being provided or created?

Reply:
Provided, but can be both, if the input is a string I could create it. We can discuss which is the best option.

Reply:
If it's provided as a Component, then it would be linked as a SubComponent.
If it's provided as a string, then it would be created as a LocalSubComponent.